components/raftstore/src/coprocessor/mod.rs

```diff
@@ -75,12 +75,21 @@ impl<'a> ObserverContext<'a> {
     }
 }
 
+/// Context of a region provided for observers.
+#[derive(Default, Clone)]
 pub struct RegionState {
     pub peer_id: u64,
     pub pending_remove: bool,
     pub modified_region: Option<Region>,
 }
 
+/// Context for exec observers of mutation to be applied to ApplyContext.
+pub struct ApplyCtxInfo<'a> {
+    pub pending_handle_ssts: &'a mut Option<Vec<SstMetaInfo>>,
+    pub delete_ssts: &'a mut Vec<SstMetaInfo>,
+    pub pending_delete_ssts: &'a mut Vec<SstMetaInfo>,
```

Comment:
where's the code that consumes pending_delete_ssts and delete the SST files?
How can we ensure that these SST files are finally deleted?

Reply:
In TiKV's view:
`post_exec` will consumes `pending_delete_ssts`, and it will return true to suggest a immediate write to db. 
In TiFlash's view:
When all data referred in the ssts is write into TiFlash's storage, TiFlash will consume `pending_delete_ssts` and suggest a write.
If TiFlash&Proxy restart before persistence, the recovered applied_index is before IngestSst command. So we can replay this command, and thus ensuring we will not lost track of sst files.

Reply:
> In TiKV's view: `post_exec` will consumes `pending_delete_ssts`, and it will return true to suggest a immediate write to db. In TiFlash's view: When all data referred in the ssts is write into TiFlash's storage, TiFlash will consume `pending_delete_ssts` and suggest a write. If TiFlash&Proxy restart before persistence, the recovered applied_index is before IngestSst command. So we can replay this command, and thus ensuring we will not lost track of sst files.

IC. Then is there any metrics for pending_delete_ssts? We need to have a way to figure this out.

Reply:
We have a tikv_raft_applying_sst added
<img width="1013" alt="image" src="https://user-images.githubusercontent.com/5670241/184799508-c77514bc-d778-4a46-9a7e-6a6183a5aa66.png">